### DISCUSSION TITLE
AI and ML using ROCm

### DISCUSSION THREAD

**Discussion by saadrahim** [ROCm/ROCm#1834] [General]
An open thread to discuss AI and ML.

**Comment by Tim-Blokdijk** [comment]
@saadrahim, will Rocm support Stable Diffusion with Navi31/32/33?

**Comment by Tim-Blokdijk** [comment]
Ok, I presume you can't say things about RDNA3. But for the 6000 series someone wrote a guide to run stable diffusion on Rocm [here](https://www.videogames.ai/2022/11/06/Stable-Diffusion-AMD-GPU-ROCm-Linux.html).
Is this the recommended way to set up Rocm? Will this guide give the best performance?
Can AMD point people managing stable diffusion projects in the right direction on how to best support Radeon cards?
For example [with open issues like these](https://github.com/cmdr2/stable-diffusion-ui/issues/115)

**Comment by Tim-Blokdijk** [comment]
@saadrahim, while we're at it, is there a way to support [GPT-J-6B](https://huggingface.co/EleutherAI/gpt-j-6B) or even [GPT-NeoX-20B](https://blog.eleuther.ai/announcing-20b/) from the [Eleuther Project](https://www.eleuther.ai)? In inference mode, I'm not going to re-train my own network just yet. Someone at nlpcloud.com [wrote a guide ](https://nlpcloud.com/deploying-gpt-neox-20-production-focus-deepspeed.html) on how to do a multi gpu setup (with Nvidia hardware).
I would like to run such models locally with AMD hardware.

**Comment by Tim-Blokdijk** [comment]
Been a while, but I just like the dynamic discussion on AI and ML here.
I looking at buying a 7900 XTX, I'm reading that they have these dedicated AI accelerators.. how can I use them?

**Reply by Mushoz** [reply]
You can't yet. ROCm 5.5.0 is required for 7900 XTX support, and that is not yet available. I would hold off from buying right now.

**Reply by Odin234** [reply]
> Been a while, but I just like the dynamic discussion on AI and ML here. I looking at buying a 7900 XTX, I'm reading that they have these dedicated AI accelerators.. how can I use them?

Just as stated by @Mushoz, ROCm 5.5.0 is required. This is only information we get from @saadrahim. We do not yet receive any updates when it will be released to public. If you are going to use it for machine learning immediately, it is not ready, take the team green instead.

But, If you are a AMD sentiment like me, and it is no problem for you to wait in uncertainty, just take that beast and wait for support.

**Reply by arch-user-france1** [reply]
You could use OpenCL implementations, however the one I found for PyTorch doesn't even run language models yet, because there's some layer missing, and it's crashing or getting stuck quite often.

From experience, I can tell you this GPU is not yet suitable for machine learning.

**Reply by Tim-Blokdijk** [reply]
@architectomega, I've been a Linux user for too long to feel any warmth towards team green. Even though their products have been working reliably on Linux for a while. I want to root for AMD but they suffer from a "black box" development methodology and "throw it over the fence" open source collaboration process. Valve seems to been able to create some cracks within AMD with development of the Vulcan and Direct3D driver stack for SteamOS/SteamDeck. Now I do want to be supportive, we're on GitHub and willing to engage on AI and ML.

It's team blue that has my attention as a real contender, they have a fully open-source driver stack that is mainline. PyTorch and TensorFlow seem to work on Arc. If it's stable and has a good price/performance I could just take the plunge.

The wildcard would be team gray (Tenstorrent) with their Grayskull accelerators.

**Reply by Odin234** [reply]
@Tim-Blokdijk Yes, I think I will look at team blue too, or considering team green. I changed my mind. It's completely nonsense to have sentiment for AMD. They are neglecting user's feedback. No communications. Still no ROCm support for RX 7900 XTX. I won't wait a millennium for this to work.
I decided to stop supporting AMD nonsense.

**Comment by Tim-Blokdijk** [comment]
George Hotz (GeoHot) was doing Gods work by publicly working trough the AMD drivers. He gave up after 6+ months of work.
https://www.youtube.com/watch?v=Mr0rWJhv9jU

**Reply by Tim-Blokdijk** [reply]
Ok, apparently Lisa Su herself contacted George and he's going to stick with AMD a little longer.
See yesterday's stream: https://www.youtube.com/watch?v=rzb2cuT9vaY

**Reply by pugzly** [reply]
It's not worth the effort. I've been waiting YEARS for some real signs of improvement, real and solid improvement, not just drips and crumbs of development from team of two interns.
Now, having to chose which team I should commit to, for the next few years, I took one last stroll through all the info form AMD's current state of support and future plans - it's a dead horse.
The way things been going, most probably Intel's GPUs and accompanying software stack - will be leaps and bound ahead of AMD within 1-2 years.
So I'm going green, it will save me tons of money and tons of time over the coming years.

**Reply by johnnynunez** [reply]
Release is coming around September. See Pull request in MiOpen.
